services/collectd/config.go

```diff
@@ -128,3 +129,37 @@ func (c *Config) Validate() error {
 
 	return nil
 }
+
+// Configs wraps a slice of Config to aggregate diagnostics.
+type Configs []Config
+
+// Diagnostics returns one set of diagnostics for all of the Configs.
+func (c Configs) Diagnostics() (*diagnostics.Diagnostics, error) {
+	d := &diagnostics.Diagnostics{
+		Columns: []string{"enabled", "bind-address", "database", "retention-policy", "batch-size", "batch-pending", "batch-timeout"},
+	}
+
+	for _, cc := range c {
+		if !cc.Enabled {
+			r := make([]interface{}, 1, len(d.Columns))
+			r[0] = false
+			d.AddRow(r)
```

Comment:
I don't recall why I thought the capacity of the row had to match the length of the header row. Changing this and the other occurrences.